Gems/Atom/RHI/Vulkan/Code/Source/RHI/Image.cpp

```diff
@@ -188,6 +618,197 @@ namespace AZ
+
+                // Queue the image's sparse binding
+                VkBindSparseInfo bindSparseInfo = m_sparseImageInfo->GetBindSparseInfo(startMip, endMip);
+                m_highestMipLevel = endMip;
+                m_residentSizeInBytes = m_sparseImageInfo->GetRequiredMemorySize(m_highestMipLevel);
+                device.GetAsyncUploadQueue().QueueBindSparse(bindSparseInfo);
+            }
+            else 
+            {
+                auto memoryView = imagePool.AllocateMemory(m_memoryRequirements.size, m_memoryRequirements.alignment);
+                if (!memoryView.IsValid())
+                {
+                    return RHI::ResultCode::OutOfMemory;
+                }
+
+                VkResult vkResult = device.GetContext().BindImageMemory(
+                    device.GetNativeDevice(), m_vkImage, memoryView.GetMemory()->GetNativeDeviceMemory(), memoryView.GetOffset());
+                AssertSuccess(vkResult);
+                result = ConvertResult(vkResult);
+                RETURN_RESULT_IF_UNSUCCESSFUL(result);
+
+                if (m_memoryView.IsValid())
+                {
+                    imagePool.DeAllocateMemory(m_memoryView);
+                }
+
+                m_memoryView = memoryView;
+                m_residentSizeInBytes = m_memoryRequirements.size;
+                m_highestMipLevel = 0;
+            }
+
+            return RHI::ResultCode::Success;
+        }
+
+        RHI::ResultCode Image::BuildSparseImage()
+        {
+            AZ_Assert(m_vkImage == VK_NULL_HANDLE, "Vulkan's native image has already been initialized.");
+
+            // Only support color sparse image for now. Support for other aspects need to added
+            if (GetAspectFlags() != RHI::ImageAspectFlags::Color)
+            {
+                return RHI::ResultCode::Fail;
+            }
+
+            auto& device = static_cast<Device&>(GetDevice());
+            const auto& physicalDevice = static_cast<const PhysicalDevice&>(device.GetPhysicalDevice());
+            const VkPhysicalDeviceFeatures& deviceFeatures = physicalDevice.GetPhysicalDeviceFeatures();
+            const RHI::ImageDescriptor& descriptor = GetDescriptor();
+
+            if (descriptor.m_dimension == RHI::ImageDimension::Image1D)
+            {
+                return RHI::ResultCode::Fail;
+            }
+
+            if (!(descriptor.m_multisampleState.m_samples == 1
+                || (descriptor.m_multisampleState.m_samples == 2 && deviceFeatures.sparseResidency2Samples)
+                || (descriptor.m_multisampleState.m_samples == 4 && deviceFeatures.sparseResidency4Samples)
+                || (descriptor.m_multisampleState.m_samples == 8 && deviceFeatures.sparseResidency8Samples)
+                || (descriptor.m_multisampleState.m_samples == 16 && deviceFeatures.sparseResidency16Samples))
+                )
+            {
+                return RHI::ResultCode::Fail;
```

Comment:
would be nice to add an  assert explaining why building a sparse image failed.

Reply:
We can't add assert. It's possible it may fail to build a sparse image. And we handle it by creating a regular vkimage.